### PR TITLE
[ADR] Against the use of Lightweight Transaction in Cassandra code base

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+(soon to be deprecated for Guice products) makes this distinction. Discussions have started on the topic and a proof of
+concept is available.
+ - `domains` we rely on LWT for throwing "AlreadyExist" exceptions. LWT are likely unnecessary as the webadmin 
+presentation layer is offering an idempotent API (and silents the AlreadyExist exceptions). Only the CLI
+(soon to be deprecated for Guice products) makes this distinction. Discussions have started on the topic and a proof of
+concept is available.
+ - `mailboxes` relies on LWT to enforce name unicity. We hit the same pitfalls than for ACLs as this is a very often
+ read table (however mailboxes of a given user being grouped together, primary key read are more limited hence this is
+ less critical). Similar results could be expected. Discussions on this topic have not been started yet. Further
+ impact studies on performance needs to be conducted.
```
Comment (Arsnael) [inline src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md:69]:
```suggestion
 impact studies on performance need to be conducted.
```

File: src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+LWT usage is required to generate `UIDs`. As append message operations tend to be limited compared to
+message update operations, this is likely less critical. UID generation could be handled via alternative systems,
+past implementations have been conducted on ZooKeeper.
+
+If not implementing IMAP CONDSTORE, generation of IMAP `MODSEQ` likely no longer makes sense. As such the fate of
+`MODSEQ` is linked to decisions on the `message` topic.
+
+Similarly, LWT are used to try to keep the count of emails in MailRepository synchronize. Such a usage is non-performance
+critical for a MDA (Mail Delivery Agent) use case but might have a bigger impact for MTA (Mail Transfer Agent). No
+discussion not work have been started on the topic.
```
Comment (Arsnael) [inline src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md:87]:
```suggestion
discussion or work have been started on the topic.
```

File: src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+Other usage of LWT includes Sieve script management, initialization of the RabbitMQMailQueue browse start and other
```
Comment (Arsnael) [inline src/adr/0044-against-the-use-of-cassandra-lightweight-transactions.md:89]:
```suggestion
Other usages of LWT include Sieve script management, initialization of the RabbitMQMailQueue browse start and other
```

**Review by Arsnael — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by rouazana — COMMENTED** [review]
>  Rely on `event sourcing` to maintain a projection of ACLs that do not rely on LWT or SERIAL consistency level. 

> LWT are required for `eventSourcing`. As event sourcing usage is limited to low-usage use cases, the performance
degradations are not an issue.

I'm probably missing your point, but for me these 2 sentences seems completely contradictory.

**Comment by rouazana** [comment]
> I'm probably missing your point, but for me these 2 sentences seems completely contradictory.

I think I got it reading your code. You assume that with event sourcing you use LWT only for writes and that you have a projection for reads.
This explanation is missing.
And then you are missing an explanation of why you cannot do that directly for every usage of LWT.
(note that I have nothing against event sourcing, and I think it's very valuable in particular for things like ACL, but your explanation is incomplete for me)

**Comment by chibenwa** [comment]
Merged

I noticed clearly that related works requires more discussions and mentionned alternative ways to potentially generate monotic integers.

**Pull request closed without merging**